crates/bevy_render/src/mesh/shape/torus.rs

```diff
@@ -27,18 +28,20 @@ impl From<Torus> for Mesh {
         // code adapted from http://apparat-engine.blogspot.com/2013/04/procedural-meshes-torus.html
         // (source code at https://github.com/SEilers/Apparat)
 
-        let n_vertices = (torus.subdivisions_segments + 1) * (torus.subdivisions_sides + 1);
-        let mut positions: Vec<[f32; 3]> = Vec::with_capacity(n_vertices);
-        let mut normals: Vec<[f32; 3]> = Vec::with_capacity(n_vertices);
-        let mut uvs: Vec<[f32; 2]> = Vec::with_capacity(n_vertices);
-
         let segment_stride = 2.0 * std::f32::consts::PI / torus.subdivisions_segments as f32;
         let side_stride = 2.0 * std::f32::consts::PI / torus.subdivisions_sides as f32;
 
-        for segment in 0..=torus.subdivisions_segments {
-            let theta = segment_stride * segment as f32;
-
-            for side in 0..=torus.subdivisions_sides {
+        let positions: Vec<[f32; 3]>;
+        let normals: Vec<[f32; 3]>;
+        let uvs: Vec<[f32; 2]>;
+
+        (positions, (normals, uvs)) = (0..=torus.subdivisions_segments)
+            .map(|segment| {
+                let theta = segment_stride * segment as f32;
+                (segment, theta)
+            })
+            .cartesian_product(0..=torus.subdivisions_sides)
```

Comment:
We've avoided the itertools dep in our tree for this long. I'd prefer to keep it it that way if this is the only justification.